routes/api/events.js

```diff
@@ -1,9 +1,376 @@
 const express = require('express');
 const router = express.Router();
+const shortid = require('shortid');
+const { check, validationResult } = require('express-validator');
+const auth = require('../../middleware/auth');
+
+const Event = require('../../models/Event');
+const User = require('../../models/User');
 
 // @route   GET api/events
-// @desc    Test route
+// @desc    Get all events
+// @access  Public
+router.get('/', async (req, res) => {
+  try {
+    const events = await Event.find();
+    res.json(events);
+  } catch (err) {
+    console.error(err.message);
+    res.status(500).send('Server Error');
+  }
+});
+
+// @route   GET api/events/m
+// @desc    Get all user's events
+// @access  Private
+router.get('/m', auth, async (req, res) => {
+  try {
+    const events = await Event.find({ user: req.user.id });
+    res.json(events);
+  } catch (err) {
+    console.error(err.message);
+    res.status(500).send('Server Error');
+  }
+});
+
+// @route   GET api/events/:id
+// @desc    Get event by ID
+// @access  Public
+router.get('/id/:id', async (req, res) => {
+  try {
+    const event = await Event.findById(req.params.id);
+
+    if (!event) {
+      return res.status(404).json({ msg: 'Event not found' });
+    }
+
+    res.json(event);
+  } catch (err) {
+    console.error(err.message);
+
+    if (err.kind === 'ObjectId') {
+      return res.status(404).json({ msg: 'Event not found' });
+    }
+
+    res.status(500).send('Server Error');
+  }
+});
+
+// @route   GET api/events/:id
+// @desc    Get all event participants
+// @access  Private
+router.get('/id/:id/participants', auth, async (req, res) => {
+  try {
+    const participants = await Event.find(
+      {
+        _id: req.params.id,
+        user: req.user.id,
+      },
+      {
+        participants: 1,
+      }
+    );
+
+    res.json(participants);
+  } catch (err) {
+    console.error(err.message);
+
+    if (err.kind === 'ObjectId') {
+      return res.status(404).json({ msg: 'Event not found' });
+    }
+
+    res.status(500).send('Server Error');
+  }
+});
+
+// @route   POST api/events
+// @desc    Create event
+// @access  Private
+router.post(
+  '/',
+  [
+    auth,
+    [
+      check('title', 'Title is required').notEmpty(),
+      check('description', 'Description is required').notEmpty(),
+      check('dateStart', 'Start date is required').notEmpty(),

```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/spumony/xvent-project-backend/pullRequest?prid=7531905)
```suggestion
      check("dateStart", 'Start date is required').notEmpty(),
```